packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputSingleSelect.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { unstable_useId as useId } from '@mui/material/utils';
+import { GridLoadIcon } from '../../icons/index';
+import { GridFilterInputValueProps } from './GridFilterInputValueProps';
+import { GridColDef } from '../../../models/colDef/gridColDef';
+
+const renderSingleSelectOptions = ({ valueOptions, valueFormatter, field }: GridColDef, api) => {
+  const iterableColumnValues =
+    typeof valueOptions === 'function'
+      ? ['', ...valueOptions({ field })]
+      : ['', ...(valueOptions || [])];
+
+  return iterableColumnValues.map((option) =>
+    typeof option === 'object' ? (
+      <option key={option.value} value={option.value}>
+        {option.label}
+      </option>
+    ) : (
+      <option key={option} value={option}>
+        {valueFormatter && option !== '' ? valueFormatter({ value: option, field, api }) : option}
+      </option>
+    ),
+  );
+};
+
+export interface GridTypeFilterInputSingleSelectProps extends GridFilterInputValueProps {
```

Comment:
We have been using the `[ComponentName]Props` convention.

```suggestion
export interface GridFilterInputSingleSelectProps extends GridFilterInputValueProps {
```